sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVDataSource.scala

```diff
@@ -118,6 +121,64 @@ object CSVDataSource {
       TextInputCSVDataSource
     }
   }
+
+  /**
+   * Checks that column names in a CSV header and field names in the schema are the same
+   * by taking into account case sensitivity.
+   */
+  def checkHeaderColumnNames(
+      schema: StructType,
+      columnNames: Array[String],
+      fileName: String,
+      enforceSchema: Boolean,
+      caseSensitive: Boolean): Unit = {
+    if (!enforceSchema && columnNames != null) {
```

Comment:
When `enforceSchema` is false and `columnName` is not null, we still should check the schema. We should issue a warning log message in this case when the schema do not match

Reply:
> When enforceSchema is false and columnName is not null ...

So, inside of the `if`, right?

> we still should check the schema.

Yes, we do the check.

> We should issue a warning log message in this case when the schema do not match

So, you suggest to replace exceptions by warning log messages or to have both?

Reply:
Probably, you mean when `enforceSchema` is `true` (we forcibly apply the schema to data), we should output warnings.